src/backend/routes/shards.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * @Author: RagingLink 
+ * @Date: 2020-07-01 21:00:54
+ * @Last Modified by: RagingLink
+ * @Last Modified time: 2020-07-01 21:07:31
+ *
+ * This project uses the AGPLv3 license. Please read the license file before using/adapting any of the code.
+ */
+
+const router = require('express').Router();
+
+router.get('/', (req, res) => {
+    res.locals.url = config.general.isbeta ? 'ws://localhost:8085' : 'wss://blargbot.xyz';
+
+    res.render('shards');
+});
+
+router.get('/json', (req, res) => {
+    res.type('json');
+    res.send(JSON.stringify(Object.values(spawner.shardCache), null, 2));
```

Comment:
Are you willing to send only shards information as an array of arrays ? Wouldn't it be better to have shards per clusters?

Reply:
https://blargbot.xyz/output/727968583769981940/raw
That's the output of `Object.values(spawner.shardCache)`, it's an array with all cluster objects, and a cluster object has a shards key with an array of shard objects

Reply:
![IMAGE](https://i.gyazo.com/082cddbf5ab9bd78a41f2b43e6ebdf3a.png)

That's quite ugly imo

Reply:
The shardCache should really be called `clusterCache`, as that's what it is.

Reply:
You could make that change too by the way :P

Preferably, I'd rather receive something like this if requesting `/shards/json`
![IMAGE](https://i.gyazo.com/98ce6202cb329177a9927d4ccc58d238.png)

Reply:
I mean, does the `shardCache` need to prettified? With `/json` you just want to get the `raw` cluster information, right?

Reply:
> You could make that change too by the way :P
> 
> Preferably, I'd rather receive something like this if requesting `/shards/json`
> ![IMAGE](https://camo.githubusercontent.com/73c16705f21533ab77de9fdc0bfd567207ee7631/68747470733a2f2f692e6779617a6f2e636f6d2f39386365363230326362333239313737613939323764346363633538643233382e706e67)

For that instead of `Object.values()` it'd just need to return `.shardCache` directly. Unless you're referring to the actual display of it?

Reply:
I think it's better to return shardCache instead of Object.values result